src/config.js

```diff
@@ -1,54 +1,76 @@
 var chalk = require('chalk');
 
+// Produce overrided configuration
 function getChalkColors(defaultConfig, overrideConfig) {
-	// make the read file understandable to chalk.
-	var chalkColors = buildChalkFunction(defaultConfig);
-	var overrideChalkColors = buildChalkFunction(overrideConfig);
-	return Object.assign({}, chalkColors, overrideChalkColors);
+	var effectiveConfig = Object.assign({}, defaultConfig, overrideConfig);
+
+	// make the effectiveConfig understandable to chalk
+	// and return it
+	return buildChalkFunction(effectiveConfig);
 }
 
 function buildChalkFunction(config) {
 	var chalkColors = {};
-	for (var color in config.colors) {
-		if (config.colors.hasOwnProperty(color)) {
+	for (var outputMethod in config) {

```

Comment:
One idea is to loop known methods and check if the configuration has a value instead of looping those provided by the configuration.... Someone can add 100 things to the configuration and make this slow for no reason...